app/[lang]/chapters/[slug]/layout.tsx

```diff
@@ -1,13 +1,23 @@
-import Navbar from 'components/Navbar'
+'use client'
+
+import Navbar, { NavbarMobile } from 'components/Navbar'
 import { chapters } from 'content'
+import useMediaQuery from 'hooks/useMediaQuery'
 
-export default async function Layout({ children, params }) {
+export default function Layout({ children, params }) {
   const isValidChapter = chapters[params.slug]
+  const isSmallScreen = useMediaQuery('(max-width: 767px)')
```

Comment:
https://user-images.githubusercontent.com/100441043/220091481-1b4e2d30-3abc-46dc-9c76-d168bdaa937d.mov

After reloading a page,` <Navbar />` appears within a few milliseconds, followed by `<NavbarMobile />` flipping. To fix this, you can use conditions in Tailwind (CSS) instead of React (JS). Here's a simple example of the code I wrote to give an idea. Class name most likely should be inside of Navbars.

```jsx
<div className='md:hidden'>
  <NavbarMobile params={params} />
</div>
<div className='hidden md:block'>
  <Navbar params={params} />
</div>
```

Reply:
Addressed in ea822132cc3d13887c40e033b9b5d1605f6ed579

Thanks a lot for the suggestion!

It made the code respond better to screen size while simplifying the code.
I went with the approach that you have suggested in the example instead of adding className `md` in individual components to keep better track of what components will be displayed and when in the same place.